tests/Unit/Domain/Test_DomainCreators.cpp

```diff
@@ -0,0 +1,177 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <algorithm>
+#include <catch.hpp>
+
+#include "DataStructures/IndexIterator.hpp"
+#include "Domain/Block.hpp"
+#include "Domain/DomainHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+// Iterates over the logical corners of a VolumeDim-dimensional cube.
+template <size_t VolumeDim>
+class CornerIterator {
```

Comment:
remove the bool has_direction and split this into two classes:  VolumeCornerIterator and FaceCornerIterator
Also add operator* to dereference